dataland-frontend/src/components/generics/ViewSingleDatasetDisplayBase.vue

```diff
@@ -1,97 +1,376 @@
+        this.processDataMetaInfoForDisplay(dataMetaInfoForNewlyChosenReportingPeriod);
+        this.routerPushToReportingPeriod(dataMetaInfoForNewlyChosenReportingPeriod.reportingPeriod);
+      } else {
+        if (newReportingPeriod) {
+          this.handleInvalidReportingPeriodPassedInUrl();
         }
+      }
+    },
+
+    /**
+     * Method to set route to a specific reporting period
+     *
+     * @param reportingPeriod Specific reporting period the route should end with
+     */
+    routerPushToReportingPeriod(reportingPeriod: string) {
+      if (this.companyId != null && this.dataType != null) {
+        this.$router
+          .push(`/companies/${this.companyId}/frameworks/${this.dataType}/reportingPeriods/${reportingPeriod}`)
+          .catch((err) =>
+            console.log("Setting route for reporting period " + reportingPeriod + " failed with error " + String(err))
+          );
+      }
+    },
+
+    /**
+     * Switches to the active dataset for the currently chosen reporting period.
+     */
+    handleClickOnSwitchToActiveDatasetForCurrentlyChosenReportingPeriodButton() {
+      this.switchToActiveDatasetForNewlyChosenReportingPeriod(
+        assertDefined(this.dataMetaInfoForDisplay?.reportingPeriod)
+      );
+    },
+
+    /**
+     * Gets the keys from a map and returns them in an alphabeticall sorted array
+     *
+     * @param inputMap The map that should be used for this operation
+     * @returns an array containing the keys of the map alphabetically sorted
+     */
+    getKeysFromMapAndReturnAsAlphabeticallySortedArray(inputMap: Map<string, object>): Array<string> {
+      return Array.from(inputMap.keys()).sort((reportingPeriodA, reportingPeriodB) => {
+        if (reportingPeriodA > reportingPeriodB) return -1;
+        else return 0;
+      });
+    },
+
+    /**
+     * Method to handle the update of the currently active data meta information for the chosen framework
+     *
+     * @param receivedMapOfReportingPeriodsToActiveDataMetaInfo 1-to-1 map between reporting periods and corresponding
+     * active data meta information objects
+     */
+    handleUpdateActiveDataMetaInfo(
+      receivedMapOfReportingPeriodsToActiveDataMetaInfo: Map<string, DataMetaInformation>
+    ) {
+      this.receivedMapOfDistinctReportingPeriodsToActiveDataMetaInfo =
+        receivedMapOfReportingPeriodsToActiveDataMetaInfo;
+      this.reportingPeriodsInDropdown = this.getKeysFromMapAndReturnAsAlphabeticallySortedArray(
+        receivedMapOfReportingPeriodsToActiveDataMetaInfo
+      );
+      this.chooseDataMetaInfoForDisplayedDataset().catch((err) =>
+        console.log("Retrieving data meta info failed with error " + String(err))
+      );
+      this.isWaitingForDataIdToDisplay = false;
+    },
+
+    /**
+     * TODO adjust: Displays either the data set using the ID from the query param or if that is not available the first data set from the list of received data sets.
```

Comment:
Todo here

Reply:
Fixed => I added a detailed jsdoc here and delted this TODO